akka-typed-tests/src/test/java/akka/typed/javadsl/ActorCompile.java

```diff
@@ -44,10 +46,33 @@ public MyMsgB(String greeting) {
         return immutable((ctx2, msg2) -> {
           if (msg2 instanceof MyMsgB) {
             ((MyMsgA) msg).replyTo.tell(((MyMsgB) msg2).greeting);
+
+            @SuppressWarnings("unused")
+            ActorRef<String> adapter = ctx2.spawnAdapter(s -> new MyMsgB(s.toUpperCase()));
           }
           return same();
         });
       } else return unhandled();
     });
   }
+
+
+  static class MyBehavior extends ExtensibleBehavior<MyMsg> {
+
+    @Override
+    public Behavior<MyMsg> receiveSignal(ActorContext<MyMsg> ctx, Signal msg) throws Exception {
+      return this;
+    }
+
+    @Override
+    public Behavior<MyMsg> receiveMessage(ActorContext<MyMsg> ctx, MyMsg msg) throws Exception {
+      // FIXME this doesn't work with Scala 2.12 "reference to spawnAdapter is ambiguous"
+      @SuppressWarnings("unused")
+      ActorRef<String> adapter = ctx.spawnAdapter(s -> new MyMsgB(s.toUpperCase()));
```

Comment:
Here is one remaining issue. It's because `akka.typed.ActorContext` implements both javadsl and scaladsl.
I'm thinking that we should make `akka.typed.ActorContext` empty, with only `asJava` and `asScala` methods. It should be very seldom users need to care about `akka.typed.ActorContext`, e.g. when implementing `ExtensibleBehavior` like this.